GiniVision/Classes/ImageMetaInformationManager.swift

```diff
@@ -255,11 +265,49 @@ internal struct ImageMetaInformationManager {
         return nil
     }
     
-    fileprivate func userComment() -> String {
+    fileprivate func userComment(rotationDegrees:Int? = nil) -> String {
         let platform = "iOS"
         let osVersion = UIDevice.current.systemVersion
         let giniVisionVersion = GiniVision.versionString
-        return "Platform=\(platform),OSVer=\(osVersion),GiniVisionVer=\(giniVisionVersion)"
+        let uuid = imageUUID()
+        var comment = "Platform=\(platform),OSVer=\(osVersion),GiniVisionVer=\(giniVisionVersion),\(userCommentUUID)=\(uuid)"
+        if let rotationDegrees = rotationDegrees {
+            // normalize the rotation to 0-360
+            let rotation = imageRotation() + rotationDegrees
+            let rotationNorm = normalizedDegrees(imageRotation: rotation)
+            comment += ",\(userCommentRotation)=\(rotationNorm)"
+        }
+        return comment
+    }
+    
+    fileprivate func imageUUID() -> String {
+        // if it already has one, reuse it - it shouldn't change
+        // if not, generate it
+        let existingUUID = uuidFromImage()
+        return existingUUID ?? NSUUID().uuidString
+    }
+    
+    fileprivate func imageRotation() -> Int {
```

Comment:
I think the function's name should reflect its intent more clearly. Maybe `imageRotationDelta`?